pkgs/by-name/to/tone/package.nix

```
@@ -1,33 +1,46 @@
-{ lib, fetchFromGitHub, buildDotnetModule, ffmpeg-full, dotnetCorePackages }:
+{
+  lib,
+  fetchFromGitHub,
+  buildDotnetModule,
+  ffmpeg-full,
+  stdenv,
+  versionCheckHook,
+}:
 
 buildDotnetModule rec {
   pname = "tone";
-  version = "0.1.5";
+  version = "0.2.3";
 
   src = fetchFromGitHub {
     owner = "sandreas";
-    repo = pname;
+    repo = "tone";
     rev = "v${version}";
-    sha256 = "sha256-HhXyOPoDtraT7ef0kpE7SCQbvGFLrTddzS6Kdu0LxW4=";
+    hash = "sha256-fe0fcX617vsRdu42FEan181Zh5Todt9FLtOuD+C7yqA=";
   };
 
   projectFile = "tone/tone.csproj";
   executables = [ "tone" ];
-  nugetDeps = ./nuget-deps.nix;
+  nugetDeps = ./deps.nix;
 
   dotnetInstallFlags = [
     "-p:PublishSingleFile=false"
   ];
 
-  dotnet-sdk = dotnetCorePackages.sdk_6_0;
   runtimeDeps = [ ffmpeg-full ];
 
-  meta = with lib; {
+  doInstallCheck = true;
+  nativeInstallCheckInputs = [ versionCheckHook ];
+
+  passthru.updateScript = ./update.sh;
+
+  meta = {
     homepage = "https://github.com/sandreas/tone";
     description = "Cross platform utility to dump and modify audio metadata for a wide variety of formats";
-    license = licenses.asl20;
-    maintainers = [ maintainers.jvanbruegge ];
-    platforms = platforms.linux;
+    changelog = "https://github.com/sandreas/tone/releases/tag/v${version}";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ jvanbruegge jwillikers ];
+    platforms = with lib.platforms; linux ++ darwin ++ windows;
+    broken = !stdenv.hostPlatform.isLinux;
```

Interesting pattern, and make sense since upstream supports these platforms but we only tested the linux build, but I haven't seen packages being marked as broken without someone observing the failure.

Nit:
```suggestion
    platforms = with lib.platforms; linux ++ darwin ++ windows;
    broken = !stdenvNoCC.isLinux;
```

---

Done. I think that `platforms` is supposed to represent all the platforms a package supports. I don't really have a good way test on a mac, and I don't even want to think about trying to run Nix on Windows, which is why I did it this way. I was recommended this approach when submitting either the `wlink` or `wchisp` package, since Linux was all I'd confirmed as working.

---

I just tested this on both darwin archs, and it works at least enough to do `--help` and `--version`.

I think we should remove `broken` and let it be added if/when actual problems are discovered.